### DISCUSSION TITLE
Target a full battery by and of day, not earlier.

### DISCUSSION THREAD

**Discussion by ndrinta** [nathanmarlor/foxess_em#28] [Ideas]
For the same reasons I exposed in #26 I thought about a different charging profile during solar production, but this time is trickier.

Looking at a generic day (I purposely picked a normal day without weird sun behaviours that would likely generate exceptions, for now I'm just thinking out loud, then we can think about all sorts of edge cases) you can see that the forecast graph in the Solar production view is generated interpolating of a series of points, knowing the value of each point would open a lot of possibilities.

![image](https://user-images.githubusercontent.com/4704172/197196346-1497fc17-071b-4cd1-8d5a-054cec96f309.png)

Do you know if you can fetch them somewhere?

**Comment by nathanmarlor** [comment]
Yep, the solar forecast is pulled from the Solcast API in half an hour intervals. The integration then upsamples into 1 minute intervals, essentially giving a solar forecast for each minute of the day. 

https://github.com/nathanmarlor/foxess_em/blob/6a569d83e333309df01ef5a56b71c7750a719ac8/custom_components/foxess_em/forecast/forecast_model.py#L77
